src/vsr/replica.zig

```diff
@@ -7245,6 +7246,16 @@ pub fn ReplicaType(
                 });
                 return;
             }
+            // To avoid falsely contributing to the durability of the current checkpoint, replicas
+            // syncing table blocks should not send prepare_oks for prepares past (and including)
+            // the next checkpoint's trigger.
+            if (!self.sync_content_done() and header.op >= self.op_checkpoint_next_trigger()) {
```

Comment:
Hm, I need to revisit this once I stop paniking over my currently deeply in progress slides, but one subtelty which is I am not sure if taker care of here is that the current checkpoint changes.

Imagine a scenario where:

* we sync to checkpoint X
* we avoid externalizing prepare_ok which signal that our _current_ checkpoint is durable
* non-the-less, we continue commiting normaly, and move to checkpoint `X + checkpoint_inteval`
* this _doesnt_ mean that our state sync finished! Even though we have enough of `X` to commit up to the next checkpoint, there might be some data which simply wasn't acceded yet. 
* and yet, because we moved to the next checkpoint, _now_ will be sending prepare oks that we were witholding previously

My original thinking here is that, while the sync isn't done, we'll have to withold prepare oks around _every_ checkpoint trigger. 

---

Another surprising thing is that we use `op_checkpoint_next_trigger` here. I'd expect us using

```
                const prepare_max_for_current_checkpoint =
                    vsr.Checkpoint.prepare_max_for_checkpoint(self.op_checkpoint()).?;
```

to mirror the logic we have in op_repair_min

But overall super low-confidence on both point: I'll need to actually think about this to be sure, and I am avoiding that, otherwise I won't do my slides on time :stuck_out_tongue:

Reply:
I will ponder on these while you work on your slides! 🤔 🕵🏼